pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -69,7 +74,25 @@ func newAttachment(
 		}
 	}()
 
-	a.collection, err = ebpf.NewCollection(spec)
+	spec = spec.Copy()
+
+	u32netns := uint32(netns)
+	consts := map[string]interface{}{
+		"current_netns": u32netns,
+	}
+
+	if err := spec.RewriteConstants(consts); err != nil && !strings.Contains(err.Error(), "spec is missing one or more constants") {
```

Comment:
Done: https://github.com/cilium/ebpf/pull/902

Reply:
Merged via https://github.com/cilium/ebpf/pull/904

Reply:
I will first get the new cilium/ebpf via a separate PR: https://github.com/inspektor-gadget/inspektor-gadget/pull/1274

Reply:
I tried to update this code to use the `MissingConstantsError` type introduced in cilium/ebpf, however it's not that simple as it is defined as a structure, and then I'll have to use `errors.As()`. I was able to make it work in the end, but then I realized this approach doesn't work good as the map replacements logic will fail too:

```bash
$ sudo ./local-gadget trace dns
WARN[0000] Runtime enricher (cri-o): couldn't get current containers 
CONTAINER                                   QR NAMESERVER              TYPE      QTYPE      NAME                                        RCODE                 
err: container : start tracing container "local-registry": creating network tracer attachment for pid 3389: failed to create BPF collection: replacement map sockets not found in CollectionSpec
```

I think we have to change the strategy, first check if the given tracer is compiled with socket enricher support and then based on that replace the constants and the maps. I implemented this in https://github.com/inspektor-gadget/inspektor-gadget/commit/a9cb529acf8c020aa0cdcb1abea5ac8b1566a06e, I think it's fine to look for a map called `sockets`, if we think this could conflicts then we should switch the whole map name to something else.